bin/nic.pl

```diff
@@ -183,6 +187,23 @@
 	}
 	untie(@lines);
 }
+
+if(!$isSubproject) {
+	promptIfMissing(\$requestsGit, "Y", "Initialise a Git repository for the project");
+	$requestsGit = (uc($requestsGit) eq "N") ? 0 : 1;
+
+	if($requestsGit) {
+		chdir($dirname);
+		my $init = `git init; git add -A; git commit -m "Initial Commit"`;
+	}
+} else {
+	# Probably don't want to prompt for git in subproject. Only if user has already requested
+	# git in the config do we add this commit message otherwise we don't know if git is setup
+	if($requestsGit) {
```

Comment:
Perhaps you could also run `git add` if the main project already has a git repository.

Reply:
i.e. if `$requestsGit` is true or `.git/` exists